File: web/src/data/calendarData.ts
// Local demo data for the Calendar screen (May 2024). Mirrors the reference.

export type DotColor = 'green' | 'amber' | 'red'

export interface DayCell {
  day: number
  inMonth: boolean
  today?: boolean
  dots?: DotColor[]
}

// May 2024 starts on Wednesday. Grid is Sun..Sat.
export const monthLabel = 'May 2024'
export const weekdays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']

const g: DotColor[] = ['green']
const rawDays: DayCell[] = [
  { day: 28, inMonth: false },
  { day: 29, inMonth: false },
  { day: 30, inMonth: false },
  { day: 1, inMonth: true, dots: g },
  { day: 2, inMonth: true, dots: g },
  { day: 3, inMonth: true, dots: ['amber', 'red'] },
  { day: 4, inMonth: true },
  { day: 5, inMonth: true, dots: g },
  { day: 6, inMonth: true, dots: g },
  { day: 7, inMonth: true, dots: ['red'] },
  { day: 8, inMonth: true, dots: g },
  { day: 9, inMonth: true, dots: ['amber'] },
  { day: 10, inMonth: true, dots: g },
  { day: 11, inMonth: true },
  { day: 12, inMonth: true, dots: g },
  { day: 13, inMonth: true, dots: g },
  { day: 14, inMonth: true, dots: g },
  { day: 15, inMonth: true, dots: g },
  { day: 16, inMonth: true, dots: ['amber'] },
  { day: 17, inMonth: true, dots: g },
  { day: 18, inMonth: true, dots: g },
  { day: 19, inMonth: true, dots: g },
  { day: 20, inMonth: true, today: true, dots: g },
  { day: 21, inMonth: true, dots: g },
  { day: 22, inMonth: true, dots: ['red'] },
  { day: 23, inMonth: true, dots: g },
  { day: 24, inMonth: true, dots: g },
  { day: 25, inMonth: true, dots: g },
  { day: 26, inMonth: true, dots: g },
  { day: 27, inMonth: true, dots: ['amber'] },
  { day: 28, inMonth: true, dots: g },
  { day: 29, inMonth: true, dots: g },
  { day: 30, inMonth: true, dots: g },
  { day: 31, inMonth: true, dots: g },
  { day: 1, inMonth: false },
]

// Empty by default — no adherence dots until the user logs doses.
export const days: DayCell[] = rawDays.map((d) => ({ day: d.day, inMonth: d.inMonth, today: d.today }))
// Suppress unused-var noise from the seed helper.
void g

export type TimeOfDay = 'morning' | 'noon' | 'night'
export interface DayMed {
  id: string
  name: string
  detail: string
  time: string
  when: TimeOfDay
  status: 'taken' | 'late' | 'missed'
}

export const selectedDayLabel = 'Monday, 20 May 2024'
export const selectedDayStats = { taken: 0, late: 0, missed: 0 }
export const selectedDayMeds: DayMed[] = []

export const quickStats = {
  month: 'May 2024',
  takenDays: 0,
  lateDays: 0,
  missedDays: 0,
  adherence: 0,
}

File: ios/MediCenter/Features/Calendar/CalendarData.swift
import SwiftUI

// Local demo data for the Calendar screen (May 2024). Mirrors the web app.

enum DotColor { case green, amber, red }

struct DayCell: Identifiable {
    let id = UUID()
    let day: Int
    let inMonth: Bool
    var today: Bool = false
    var dots: [DotColor] = []
}

enum TimeOfDay { case morning, noon, night }

struct DayMed: Identifiable {
    let id = UUID()
    let name: String
    let detail: String
    let time: String
    let when: TimeOfDay
    let status: DoseStatus
}

enum CalendarData {
    static let monthLabel: String = {
        let f = DateFormatter(); f.dateFormat = "MMMM yyyy"; return f.string(from: Date())
    }()
    static let weekdays = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]

    /// Real month grid for the current month, with today marked and no fake status dots.
    static let days: [DayCell] = {
        let cal = Calendar.current
        let now = Date()
        let comps = cal.dateComponents([.year, .month], from: now)
        guard let first = cal.date(from: comps),
              let range = cal.range(of: .day, in: .month, for: now) else { return [] }
        let leading = (cal.component(.weekday, from: first) - 1) // Sun-based
        let todayDay = cal.component(.day, from: now)
        var cells: [DayCell] = []
        // leading days from previous month (greyed)
        if leading > 0 {
            for i in stride(from: leading, to: 0, by: -1) {
                if let d = cal.date(byAdding: .day, value: -i, to: first) {
                    cells.append(DayCell(day: cal.component(.day, from: d), inMonth: false))
                }
            }
        }
        for day in range {
            cells.append(DayCell(day: day, inMonth: true, today: day == todayDay))
        }
        // trailing to fill the last week
        var trailing = 1
        while cells.count % 7 != 0 { cells.append(DayCell(day: trailing, inMonth: false)); trailing += 1 }
        return cells
    }()

    static var selectedDayLabel: String {
        let f = DateFormatter(); f.dateFormat = "EEEE, d MMM yyyy"; return f.string(from: Date())
    }

    static var selectedStats: (taken: Int, late: Int, missed: Int) { (0, 0, 0) }

    /// Today's medicines, synced from the user's store.
    static var selectedMeds: [DayMed] {
        MedStore.shared.medications
            .sorted { HomeData.minutes($0.time) < HomeData.minutes($1.time) }
            .map { med in
                let hour = HomeData.minutes(med.time) / 60
                let when: TimeOfDay = hour < 12 ? .morning : (hour < 17 ? .noon : .night)
                return DayMed(name: med.name, detail: "\(med.dose) · \(med.food)", time: med.time, when: when, status: .upcoming)
            }
    }

    static var quick: (month: String, takenDays: Int, lateDays: Int, missedDays: Int, adherence: Int) {
        (monthLabel, 0, 0, 0, 0)
    }
}
